### PR TITLE
[Fixes #4317] UrlResolutionTagHelper doesn't work with TagHelpers that run before and supress output

### LINKED ISSUE #4317: UrlResolutionTagHelper doesn't work with TagHelpers that run before and supress output.
If a user has a `TagHelper` and sets its order to `int.MinValue`, it'll run before the `UrlResolutionTagHelper`. Now, if that same `TagHelper` calls `output.SuppressOutput()` it will then force the `TagHelperOutput`s tag name to `null`. The problem this introduces is the `UrlResolutionTagHelper` relies on the `TagName` not being `null` when looking up corresponding attribute values for the tag so it ends up exploding [here](https://github.com/aspnet/Mvc/blob/fd984563c3f7e32c41a37c2d21040b6e4490d39c/src/Microsoft.AspNetCore.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs#L129).

We should check if `TagName == null` and if so, bail early.

### PR DESCRIPTION
**Pull request by javiercn**
Check if the TagHelperOutput.TagName is null and do nothing in that case.

### CODE REVIEW

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs [hunk 1]
```diff
@@ -36,6 +36,46 @@ public static TheoryData ResolvableUrlData
...
+            // Arrange
+            var tagHelperOutput = new TagHelperOutput(
+                tagName: null,
+                attributes: new TagHelperAttributeList
+                {
+                    { "href", "~/home/index.html" }
+                },
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(null));
+
+            var urlHelperMock = new Mock<IUrlHelper>();
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs]: No need for this guy.

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs [hunk 2]
```diff
@@ -36,6 +36,46 @@ public static TheoryData ResolvableUrlData
...
+            urlHelperMock
+                .Setup(urlHelper => urlHelper.Content(It.IsAny<string>()))
+                .Returns(new Func<string, string>(value => "/approot" + value.Substring(1)));
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs]: `.Throws`

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs [hunk 3]
```diff
@@ -36,6 +36,46 @@ public static TheoryData ResolvableUrlData
...
+
+            var urlHelperFactory = new Mock<IUrlHelperFactory>();
+            urlHelperFactory
+                .Setup(f => f.GetUrlHelper(It.IsAny<ActionContext>()))
+                .Returns(urlHelperMock.Object);
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs]: `.Throws`

File: test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs [hunk 4]
```diff
@@ -36,6 +36,46 @@ public static TheoryData ResolvableUrlData
...
+                    { "href", "~/home/index.html" }
+                },
+                getChildContentAsync: (useCachedResult, encoder) => Task.FromResult<TagHelperContent>(null));
+
+            var urlHelperMock = new Mock<IUrlHelper>();
+            urlHelperMock
+                .Setup(urlHelper => urlHelper.Content(It.IsAny<string>()))
+                .Returns(new Func<string, string>(value => "/approot" + value.Substring(1)));
+
+            var urlHelperFactory = new Mock<IUrlHelperFactory>();
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Mvc.Razor.Test/TagHelpers/UrlResolutionTagHelperTest.cs]: `MockBehavior.Strict`. Can remove the rest.

**Comment by NTaylorMullen** [comment]
:watch: for test changes.

**Comment by javiercn** [comment]
Updated sorry for the accidnental rebase

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request closed without merging**